File: 백준/Silver/10828. 스택/스택.py
import sys

n = int(input())
stack = list()

size = 0

for _ in range(n):
    contex=sys.stdin.readline().split()

    if contex[0] == "push":
        stack.append(contex[1])
        size += 1
    elif contex[0] == "top":
        if size == 0:
            print(-1)
        else:
            print(stack[size - 1])
    elif contex[0] == "size":
        print(size)
    elif contex[0] == "empty":
        if size == 0:
            print(1)
        else:
            print(0)
    elif contex[0] == "pop":
        if size == 0:
            print(-1)
        else:
            print(stack.pop())
            size -= 1

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>

using namespace std;

int N;
stack<int> s;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    cin >> N;
    for(int i=0;i<N;i++) {
        string a;
        cin >> a;
        
        if(a=="push") {
            int b;
            cin >> b;
            s.push(b);
        }
        else if(a=="pop") {
            if(s.size()==0) {
                cout << -1 << "\n";
            }
            else {
                cout << s.top() << "\n";
                s.pop();
            }
        }
        else if(a=="size") {
            cout << s.size() << "\n";
        }
        else if(a=="empty") {
            if(s.empty()) {
                cout << 1 << "\n";
            }
            else {
                cout << 0 << "\n";
            }
        }
        else if(a=="top") {
            if(s.size()==0) {
                cout << -1 << "\n";
            }
            else {
                cout << s.top() << "\n";
            }
        }
    }
    
    return 0;
    
}
